### PR TITLE
Improve pinning/unpinning in Priority Groups

### PR DESCRIPTION
**Pull request by Jarema**
After round of feedback, few changes were requested:

**Fix unpin advisory name**

Copy-paste mistake was made in unpin advisory name.

**Fix PinnedTS**

It was meant to return time when pinned happened, not the daedline.

**Add check for group name**

As the ADR for Priority Groups was [also updated](https://github.com/nats-io/nats-architecture-and-design/pull/311) after feedbach, this PR implements that.

cc @ripienaar @jnmoyne 

Signed-off-by: Tomasz Pietrek <tomasz@nats.io>

### CODE REVIEW

**Review by ripienaar — APPROVED** [review]
LGTM

File: server/consumer.go [hunk 1]
```diff
@@ -818,6 +823,9 @@ func checkConsumerCfg(
 			if group == _EMPTY_ {
 				return NewJSConsumerEmptyGroupNameError()
```
Comment (derekcollison) [inline server/consumer.go:824]: Is this one used anywhere? Should we remove if not?

Comment (Jarema) [reply server/consumer.go:824]: you mean - remove the error, as the regex below will check for empty groups?

Comment (derekcollison) [reply server/consumer.go:824]: ok if still used we are good, could not tell if the empty error was being used anymore.

**Pull request merged**